streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -577,28 +575,35 @@ private void runLoop() {
                     polledRecords = false;
                 }
 
-                // only record poll latency is long poll is required
-                if (longPoll) {
-                    streamsMetrics.pollTimeSensor.record(computeLatency(), timerStartedMs);
-                }
+                streamsMetrics.pollTimeSensor.record(computeLatency(), timerStartedMs);
             }
 
-            // try to process one fetch record from each task via the topology, and also trigger punctuate
+            // try to process several records from each task via the topology, and also trigger punctuate
             // functions if necessary, which may result in more records going through the topology in this loop
-            if (totalNumBuffered > 0 || polledRecords) {
-                totalNumBuffered = 0;
-
+            if (polledRecords) {
```

Comment:
I think we can probably get rid of `polledRecords` and move the `if(!activeTasks.isEmpty()) {...}` up into the previous `if(records != null && !records.isEmpty()){..}`